### ISSUE TITLE
Rancher v2.3.5 pipeline not working with gitlab

### ISSUE THREAD

**Issue by bolotkalil** [rancher/rancher#25657]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**

- Goto rancher pipeline section
- Choose gitlab as provider of repositories
- Fill Application Id, Secret and click Authenticate 
- Got this error: 
"**TestAndApply Error
Fail to get accesstoken with oauth config**"
![rancher_gitlab_pipeline_bug](https://user-images.githubusercontent.com/4947951/75551620-d11d3780-5a5e-11ea-84b7-da80da807192.png)

**Expected Result:**
List of available repositories

**Other details that may be helpful:**
- Other providers github and bitbucket work fine.
- Credentials, redirection URI and grant permissions are correct.
- When I run this curl request: 
```
curl -u "${CATTLE_ACCESS_KEY}:${CATTLE_SECRET_KEY}" \
-X POST \
-H 'Accept: application/json' \
-H 'Content-Type: application/json' \
-d '{"clientId":"[application_id]", "clientSecret":"[secret]", "code":"[code]", "hostname":"gitlab.com", "redirectUrl":"https://node1.example.com/verify-auth", "tls":true}' \
'https://node1.example.com/v3/project/local:p-2z52v/gitlabPipelineConfigs/p-2z52v:gitlab?action=testAndApply'
```
_I got this response:_

```
oauth2: cannot fetch token: 401 Unauthorized\nResponse: {\"error\":\"invalid_grant\",\"error_description\":\"The provided authorization grant is invalid, expired, revoked, does not match the redirection URI used in the authorization request, or was issued to another client.\"}
```
**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):rancher/rancher:v2.3.5
- Installation option (single install/HA):HA(rke)

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):Imported
- Machine type (cloud/VM/metal) and specifications (CPU/memory):DO Droplet 
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.1", GitCommit:"d647ddbd755faf07169599a625faf302ffc34458", GitTreeState:"clean", BuildDate:"2019-10-02T17:01:15Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-13T11:13:49Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
Client:
 Version:      1.10.3
 API version:  1.24
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 21:49:11 2016
 OS/Arch:      linux/amd64
```

**Comment by zgc** [comment]
+1

**Comment by easonjim** [comment]
+1

**Comment by easonjim** [comment]
This is a problem with gitlab version, it is ok to use 12.4.x version, do not use the latest version. I now use version 2.3.5 for rancher.

**Comment by easonjim** [comment]
I now find that there are many problems with the latest version of gitlab's docker image, such as laster version 12.8.x cannot be restarted and so on.

**Comment by tschmi5** [comment]
+1

**Comment by deniseschannon** [comment]
This is fixed in v2.3.6 and v2.4.0: https://github.com/rancher/rancher/issues/25975
